### PR TITLE
Update documentation ahead of merge into develop

### PR DESCRIPTION
**Pull request by WillChilds-Klein**
*Issue #, if available:* [i/CryptoAlg-936](https://sim.amazon.com/issues/CryptoAlg-936)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## 2.0.0 (Unreleased)
```
Comment (SalusaSecondus) [inline CHANGELOG.md:3]: If we're bumping this, we should probably bump the version in build.gradle to `2.0.0` as well.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+* Use [AWS-LC](https://github.com/awslabs/aws-lc/) as the backing native cryptography library for ACCP
+* Track the AWS-LC dependency as a git submodule instead of downloaded tarball
+* Drop support for (non-EC) DSA signatures
+* Use AWS-LC's DRBG implementation, drop custom java implementation
+* Drop support for `secp192r1`, as well as other non-NIST "legacy" curves
+
 ## 1.7.0 (Unreleased)
```
Comment (SalusaSecondus) [inline CHANGELOG.md:10]: Do we think that 1.7.0 will ever actually exist? If it won't, merge it into the 2.0.0 notes.

File: DEVELOPMENT-GUIDE.md [hunk 3]
```diff
@@ -140,18 +140,18 @@ Each layer from highest level (Java) down:
 Once a [Java Exception](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/design.html#java_exceptions) has been thrown, there are exceedingly few JNI operations a caller is allowed to perform.
 What's more, it is easy to not notice that there is a Java exception pending.
 So long as you do not interact with `JNIEnv`/`raii_env` directly, you shouldn't need to worry about Java exceptions as the ACCP objects/methods which use them already have appropriate checks which throw C++ exceptions (specifically `java_ex`) instead.
-If you need to throw an exception, it should almost always be a C++ exception and thrown using the `throw_java_ex` or `throw_openssl` methods. (The former is more efficient but the latter MUST be used when the exception is due to an error state reported by OpenSSL.)
```
Comment (SalusaSecondus) [inline DEVELOPMENT-GUIDE.md:143]: Let's add a note to this page that as ACCP was originally built on openssl, many functions/etc. refer to openssl when they really mean any implementation of `libcrypto` such as AWS-LC.

File: README.md [hunk 4]
```diff
@@ -233,3 +233,9 @@ This library is licensed under the Apache 2.0 license although portions of this
 Specifically, this product includes software developed by the OpenSSL Project for use in the OpenSSL Toolkit ([http://www.openssl.org](http://www.openssl.org/))
 and this product also includes cryptographic software written by Eric Young (eay@cryptsoft.com).
 
+TODO (callout for reviewers) -- what to do about licensing differences between
+OpenSSL and AWS-LC? Is there anything meaningful here? It looks like all of
+AWS-LC's antecedents are published under apache 2.0 as well:
+
+https://github.com/awslabs/aws-lc/blob/main/LICENSE#L1-L20
+
```
Comment (SalusaSecondus) [inline README.md:241]: If we are not entirely under Apache 2.0 with no remaining stuff from the old dual OpenSSL/SSLeay license, then lets remove anything in this section not relevant to Apache 2.0. (Or better, note that as of version 2.0, only Apache 2.0 applies.)

File: CHANGELOG.md (hunk 2 above)
Comment (WillChilds-Klein) [reply CHANGELOG.md:10]: i don't se  a particular reason why we'd do a 1.7.0 release at this point. I'll pull up the interesting and still-relevant bits into 2.0.0's section.

File: CHANGELOG.md (hunk 1 above)
Comment (WillChilds-Klein) [reply CHANGELOG.md:3]: yep, it [already it](https://github.com/corretto/amazon-corretto-crypto-provider/blob/aws-lc/build.gradle#L12).

File: DEVELOPMENT-GUIDE.md (hunk 3 above)
Comment (WillChilds-Klein) [reply DEVELOPMENT-GUIDE.md:143]: good call, will add.

File: README.md (hunk 4 above)
Comment (WillChilds-Klein) [reply README.md]: After reading through AWS-LC's `LICENSE` file more closely, it looks like AWS-LC [still uses](https://github.com/awslabs/aws-lc/blob/main/LICENSE#L8-L20) the combination Apache 2.0 + OpenSSL/SSLeay, as well as adding a few more licensing pieces courtesy of BoringSSL (some for source dependencies, some for test/build dependencies which aren't distributed).

I've added a sentence pointing to this change.

File: README.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 The Amazon Corretto Crypto Provider (ACCP) is a collection of high-performance cryptographic implementations exposed via the standard [JCA/JCE](https://docs.oracle.com/en/java/javase/11/security/java-cryptography-architecture-jca-reference-guide.html) interfaces.
 This means that it can be used as a drop in replacement for many different Java applications.
 (Differences from the default OpenJDK implementations are [documented here](./DIFFERENCES.md).)
-Currently algorithms are primarily backed by OpenSSL's implementations (1.1.1n as of ACCP 1.7.0) but this may change in the future.
+Currently algorithms are primarily backed by AWS-LC's implementations but this may change in the future.
```
Comment (SalusaSecondus) [inline README.md:5]: Please add an "As of version 2.0.0" or similar here.

**Review by SalusaSecondus — COMMENTED** [review]
Please ensure that that the algorithms list in the readme is still accurate. Please also update the maven dependency snippet with a comment that once 2.0.0 is release, the version line will need to be changed.

File: README.md [hunk 6]
```diff
@@ -39,10 +39,13 @@ Mac algorithms:
 
 Cipher algorithms:
 * AES/GCM/NoPadding
+* AES_128/GCM/NoPadding
+* AES_256/GCM/NoPadding
 * RSA/ECB/NoPadding
-* RSA/ECB/PKCS1Padding
+* RSA/ECB/Pkcs1Padding
```
Comment (SalusaSecondus) [inline README.md:45]: As per [Standard Algorithm Names](https://docs.oracle.com/javase/9/docs/specs/security/standard-names.html#cipher-algorithm-paddings) we should leave it capitalized as it was before.
```suggestion
* RSA/ECB/PKCS1Padding
```

File: README.md [hunk 7]
```diff
@@ -104,6 +105,7 @@ This will instruct it to use the most recent 1.x version of ACCP.
 For more information, please see [VERSIONING.rst](https://github.com/corretto/amazon-corretto-crypto-provider/blob/develop/VERSIONING.rst).
 
 ```xml
+<!-- TODO: once ACCPv2.0.0 has been released, we need to update the version field below to reflect this. -->
```
Comment (SalusaSecondus) [inline README.md:108]: Let's make this a public comment. Something like:

> The below snippet will pull in all versions of ACCP prior to the 2.0.0 release. Once that it released we recommend that everyone switch to the `[2.0,3.0)` specifier instead.

Comment (WillChilds-Klein) [reply README.md:108]: sure

**Review by SalusaSecondus — APPROVED** [review]
LGTM

**Comment by WillChilds-Klein** [comment]
thanks. merging now (i.e. before checks finish) because this commit is documentation-only -- no code changes.

**Pull request merged**